### PR TITLE
feat: Added Treeview feature

### PR DESCRIPTION
**Pull request by Darkmift**
# Description
- I started work on a treeview display of json data
- if its of any use let me know and Ill work out the visuals further
## screenshots
image

### CODE REVIEW

File: src/pages/components/map-related/MapLayers/BusToolTip.tsx [hunk 1]
```diff
@@ -94,12 +95,22 @@ export function BusToolTip({ position, icon }: BusToolTipProps) {
...
+              data={siriRide}
+              name={t('drive_direction')}
+            />
+          )}
+        </div>
+        // <pre>
+        //   {JSON.stringify(position, null, 2)}
+        //   <br />
+        //   {siriRide && JSON.stringify(siriRide, null, 2)}
+        // </pre>
```
Comment (NoamGaash) [inline src/pages/components/map-related/MapLayers/BusToolTip.tsx:112]: should it be removed?

**Review by NoamGaash — APPROVED** [review]
That's so cool! Thank you :1st_place_medal:

File: src/pages/components/map-related/MapLayers/BusToolTip.tsx (hunk 1 above)
Comment (Darkmift) [reply src/pages/components/map-related/MapLayers/BusToolTip.tsx]: Yes!

That was the previous display.
I removed it just now.

**Comment by NoamGaash** [comment]
@Darkmift did you close the PR on purpose?

**Comment by Darkmift** [comment]
> @Darkmift did you close the PR on purpose?

Hey apologies,
not sure when or how I closed this :/

**Comment by NoamGaash** [comment]
Hi, I'm sorry - seems like `npm run lint` doesn't pass on this branch.
I'll fix the CI pipeline, and revert the changes,
Let's open a new PR with a valid version

Please try to avoid `any`, and to explicitly tell what types are expected for each of the functions and variables. It could help other developers to get a better understanding of what you did

**Comment by NoamGaash** [comment]
I've opened [this pr](https://github.com/hasadna/open-bus-map-search/pull/563)

**Pull request merged**